keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -162,6 +163,17 @@ interface ArtifactRepository : PeriodicallyCheckedRepository<DeliveryArtifact> {
     targetEnvironment: String
   )
 
+  /**
+   * Marks a version of an artifact as skipped for an environment, with information on what version skipped it.
+   */
+  fun markAsSkipped(
```

Comment:
This function is currently only used in tests. Do you anticipate it being used elsewhere? If not, should probably move it to the test classes.

Reply:
I anticipate that it might, which is why I kept it here.